test/EFCore.PG.FunctionalTests/Query/EnumQueryTest.cs

```diff
@@ -130,12 +130,47 @@ public void Enum_ToString()
             var _ = ctx.SomeEntities.Single(e => e.MappedEnum.ToString().Contains("sa"));
 
             AssertSql(
-                @"SELECT s.""Id"", s.""EnumValue"", s.""InferredEnum"", s.""MappedEnum"", s.""SchemaQualifiedEnum"", s.""UnmappedEnum""
+                @"SELECT s.""Id"", s.""ByteEnum"", s.""EnumValue"", s.""InferredEnum"", s.""MappedEnum"", s.""SchemaQualifiedEnum"", s.""UnmappedByteEnum"", s.""UnmappedEnum""
 FROM test.""SomeEntities"" AS s
 WHERE STRPOS(CAST(s.""MappedEnum"" AS text), 'sa') > 0
 LIMIT 2");
         }
 
+        [Fact]
+        public void Where_byte_enum_array_contains_enum()
+        {
+            using var ctx = CreateContext();
+            var values = new[] { ByteEnum.Sad };
+            var result = ctx.SomeEntities.Single(e => values.Contains(e.ByteEnum));
+            Assert.Equal(2, result.Id);
+
+            AssertSql(
+                @"@__values_0='Npgsql.EntityFrameworkCore.PostgreSQL.Query.EnumQueryTest+ByteEnum[]' (DbType = Object)
+
+SELECT s.""Id"", s.""ByteEnum"", s.""EnumValue"", s.""InferredEnum"", s.""MappedEnum"", s.""SchemaQualifiedEnum"", s.""UnmappedByteEnum"", s.""UnmappedEnum""
+FROM test.""SomeEntities"" AS s
+WHERE s.""ByteEnum"" = ANY (@__values_0)
+LIMIT 2");
+        }
+
+        [Fact]
+        public void Where_unmapped_byte_enum_array_contains_enum()
+        {
+            using var ctx = CreateContext();
+            var values = new[] { UnmappedByteEnum.Sad };
```

Comment:
Looks like type mapping doesn't make you happy, sad.

Reply:
You have no idea :cry: